stl/inc/cmath

```diff
@@ -1272,96 +1275,67 @@ _NODISCARD /* constexpr */ _Ty _Common_lerp(const _Ty _ArgA, const _Ty _ArgB, co
         return _Candidate;
     }
 
-    if (isnan(_ArgA)) {
-        return _ArgA;
-    }
-
-    if (isnan(_ArgB)) {
-        return _ArgB;
-    }
-
-    if (isnan(_ArgT)) {
-        return _ArgT;
-    }
-
-    switch (_Finite_mask) {
-    case 0b000:
-        // All values are infinities
-        if (_ArgT >= 1) {
-            return _ArgB;
-        }
-
-        return _ArgA;
-    case 0b010:
-    case 0b100:
-    case 0b110:
-        // _ArgT is an infinity; return infinity in the "direction" of _ArgA and _ArgB
-        return _ArgT * (_ArgB - _ArgA);
-    case 0b001:
-        // Here _ArgA and _ArgB are infinities
-        if (_ArgA == _ArgB) {
-            // same sign, so T doesn't matter
-            return _ArgA;
-        }
-
-        // Opposite signs, choose the "infinity direction" according to T if it makes sense.
-        if (_ArgT <= 0) {
+    if (_STD is_constant_evaluated()) {
+        if (_STD _Is_nan(_ArgA)) {
             return _ArgA;
         }
 
-        if (_ArgT >= 1) {
+        if (_STD _Is_nan(_ArgB)) {
             return _ArgB;
         }
 
-        // Interpolating between infinities of opposite signs doesn't make sense, NaN
-        if constexpr (sizeof(_Ty) == sizeof(float)) {
-            return __builtin_nanf("0");
-        } else {
-            return __builtin_nan("0");
-        }
-    case 0b011:
-        // _ArgA is an infinity but _ArgB is not
-        if (_ArgT == 1) {
-            return _ArgB;
+        if (_STD _Is_nan(_ArgT)) {
+            return _ArgT;
         }
+    } else {
+        if (_STD _Is_nan(_ArgA)) {
+            // raise FE_INVALID if at least one of _ArgA, _ArgB and _ArgT is signaling NaN
+            _Ty _Tmp = _ArgA;
 
-        if (_ArgT < 1) {
-            // towards the infinity, return it
-            return _ArgA;
-        }
+            if (_STD _Is_nan(_ArgB) || _STD _Is_nan(_ArgT)) {
+                _Tmp = _ArgB + _ArgT;

```

Comment:
Not a big deal but you're potentially reassigning to `_Tmp` before ever using it here.

Reply:
Fixed and simplified